`#[rustfmt::skip]` attribute incompatible with `#[unsafe(naked)]`

The following code fails to compile due to attribute checks on naked functions being too strict.

```rust
#[rustfmt::skip]
#[unsafe(naked)]
extern "C" fn naked() {
    core::arch::naked_asm!("");
}
```

```
error[E0736]: attribute incompatible with `#[unsafe(naked)]`
 --> src/lib.rs:1:1
  |
1 | #[rustfmt::skip]
  | ^^^^^^^^^^^^^^^^ the `rustfmt::skip` attribute is incompatible with `#[unsafe(naked)]`
2 | #[unsafe(naked)]
  | ---------------- function marked with `#[unsafe(naked)]` here

For more information about this error, try `rustc --explain E0736`.
error: could not compile `playground` (lib) due to 1 previous error
```

cc @folkertdev

--- a/compiler/rustc_passes/src/check_attr.rs
+++ b/compiler/rustc_passes/src/check_attr.rs
@@ -683,7 +683,9 @@ impl<'tcx> CheckAttrVisitor<'tcx> {
                         }
                     }
 
-                    if !other_attr.has_any_name(ALLOW_LIST) {
+                    if !other_attr.has_any_name(ALLOW_LIST)
+                        && !matches!(other_attr.path().as_slice(), [sym::rustfmt, ..])
+                    {
                         let path = other_attr.path();
                         let path: Vec<_> = path.iter().map(|s| s.as_str()).collect();
                         let other_attr_name = path.join("::");

--- a/tests/ui/asm/naked-functions.rs
+++ b/tests/ui/asm/naked-functions.rs
@@ -231,3 +231,9 @@ pub extern "C" fn compatible_linkage() {
 pub extern "C" fn rustc_std_internal_symbol() {
     naked_asm!("", options(raw));
 }
+
+#[rustfmt::skip]
+#[unsafe(naked)]
+pub extern "C" fn rustfmt_skip() {
+    naked_asm!("", options(raw));
+}

